lib/algorithms/searches/binary_search.dart

```diff
@@ -0,0 +1,35 @@
+/// Searches [list] for [find] using the binary method
+/// This algorithm assumes that [list] is in ascending order
+///
+/// If the midpoint value matches [find], its position in [list] is returned
+/// If the midpoint value is less than [find], the search continues in the lower half of [list].
+/// If the midpoint value is greater than [find], the search continues in the upper half of [list]
+/// In this way, half of [list] is eliminated on every iteration.
+/// -1 is returned if [find] not in [list], returns index value of [find] if present in [list]
+///
+int binarySearch<B extends Comparable>(List<B> list, B find) {
+  var numItems = list.length;
+  var end = numItems - 1;
+  var start = 0;
+
+  // first compare end of current index list to beginning of index list
+  // to ensure mid variable is not invalid
+  // also automatically ends program if [list] is empty
+  while (start <= end) {
+    var mid = ((end + start) / 2).floor();
+    if (list[mid] == find) {
+      return mid;
+    }
+    // If the midpoint value is less than [find],
+    // the search continues in the lower half of [list]
+    else if (list[mid].compareTo(find) > 0) {
+      end = mid - 1;
+    }
+    //If the midpoint value is greater than [find],
+    // the search continues in the upper half of [list]
+    else {
+      start = mid + 1;
+    }
+  }
+  return -1;
+}
```

Comment:
Would be cool to have the recursive version of this search algorithm, but that can be done in another PR 👍